org.testeditor.web.dropwizard/src/main/java/org/testeditor/web/dropwizard/auth/ApiTokenAuthenticator.xtend

```diff
@@ -0,0 +1,17 @@
+package org.testeditor.web.dropwizard.auth
+
+import io.dropwizard.auth.AuthenticationException
+import io.dropwizard.auth.Authenticator
+import java.util.Optional
+
+class ApiTokenAuthenticator implements Authenticator<String, User> {
+
+	override authenticate(String credentials) throws AuthenticationException {
+		if (credentials == "42") { // dummy implementation for now
+			return Optional.of(new User('id', 'name', 'email'))
```

Comment:
Maybe we could use a different subclass of Principal here?

Reply:
As far as I know all authentification filters in a chain must return the same principal. I will have to check that.